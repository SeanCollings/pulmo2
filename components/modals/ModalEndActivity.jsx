import React, { useState } from 'react';
import { StyleSheet, View, Text, TouchableOpacity } from 'react-native';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';

import CustomModal from '.';
import { useTheme } from '../../hooks/useTheme';
import {
  OPTIONS_END_ACTIVITY_EARLY,
  OPTION_SELECT_A_REASON,
} from '../../constants/constants';
import OptionsPicker from '../OptionsPicker';

const ModalEndActivity = ({ cancelModal, confirmModal, resumeActivity }) => {
  const theme = useTheme();
  const [reason, setReason] = useState(OPTIONS_END_ACTIVITY_EARLY[0].value);

  const BackButton = () => (
    <TouchableOpacity
      testID="resumeActivity"
      onPress={resumeActivity}
      style={styles.backButton}
    >
      <MaterialCommunityIcons
        name="arrow-left"
        color={theme.BORDER}
        size={33}
      />
    </TouchableOpacity>
  );

  const opacity = theme.DARK ? 0.87 : 1;

  return (
    <CustomModal
      cancelModal={cancelModal}
      confirmModal={() => confirmModal(reason)}
      header={'Save current progress?'}
      cancelTitle="No"
      confirmTitle="Yes"
      headingColour={theme.SECONDARY}
      TopLeft={BackButton}
      animationType="fade"
    >
      <View style={styles.contentContainer}>
        <Text style={{ ...styles.contentText, color: theme.TEXT, opacity }}>
          Your current activity is not complete.
        </Text>
        <Text
          style={{
            ...styles.contentText,
            color: theme.TEXT,
            opacity,
            paddingBottom: 20,
          }}
        >
          Would you like to save it as is?
        </Text>
        <OptionsPicker
          selectedOption={reason}
          options={OPTIONS_END_ACTIVITY_EARLY}
          onChange={setReason}
          style={{
            opacity: reason === OPTION_SELECT_A_REASON.value ? 0.36 : opacity,
          }}
        />
      </View>
    </CustomModal>
  );
};

ModalEndActivity.defaulProps = {
  cancelModal: () => {},
  confirmModal: () => {},
  resumeActivity: () => {},
};

const styles = StyleSheet.create({
  contentContainer: {
    paddingTop: 10,
  },
  contentText: {
    textAlign: 'center',
    fontSize: 17,
    fontFamily: 'tit-light',
  },
  backButton: { position: 'absolute', left: 0 },
});

export default ModalEndActivity;
