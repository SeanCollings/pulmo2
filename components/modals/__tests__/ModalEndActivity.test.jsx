import React from 'react';
import { act, create } from 'react-test-renderer';
import { cleanup, fireEvent } from 'react-native-testing-library';
import ModalEndActivity from '../ModalEndActivity';

jest.useFakeTimers();

describe('ModalEndActivity - unit test', () => {
  afterEach(() => {
    cleanup();
  });

  test('should render default', async () => {
    await act(async () => {
      component = create(<ModalEndActivity animationType="none" />);
    });

    expect(component).toMatchSnapshot();
  });

  test('should handle resume activity button press', async () => {
    const resumeActivity = jest.fn();

    await act(async () => {
      component = create(
        <ModalEndActivity
          resumeActivity={resumeActivity}
          animationType="none"
        />
      );
    });

    const resumeActivityButton = component.root.find(
      ({ props }) => props.testID === 'resumeActivity'
    );

    fireEvent.press(resumeActivityButton);
    expect(resumeActivity).toHaveBeenCalled();
  });

  test('should handle confirm ernd activity button press', async () => {
    const confirmModal = jest.fn();

    await act(async () => {
      component = create(
        <ModalEndActivity confirmModal={confirmModal} animationType="none" />
      );
    });

    const confirmModalButton = component.root.find(
      ({ props }) => props.testID === 'confirmModal'
    );

    fireEvent.press(confirmModalButton);
    expect(confirmModal).toHaveBeenCalled();
  });
});
