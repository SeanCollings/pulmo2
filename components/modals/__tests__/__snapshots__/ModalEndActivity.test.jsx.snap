// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalEndActivity - unit test should render default 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(0, 0, 0, 0.5)",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    transparent={true}
    visible={true}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 12,
            "elevation": 5,
            "padding": 20,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 2.84,
            "width": "80%",
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-around",
              "width": "105%",
            }
          }
        >
          <View
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "left": 0,
                "opacity": 1,
                "position": "absolute",
              }
            }
            testID="resumeActivity"
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#CCC",
                    "fontSize": 33,
                  },
                  undefined,
                  Object {
                    "fontFamily": "material-community",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              ÔÅç
            </Text>
          </View>
          <Text
            style={
              Object {
                "color": "#002f56",
                "fontFamily": "tit-regular",
                "fontSize": 20,
                "textAlign": "center",
                "width": "80%",
              }
            }
          >
            Save current progress?
          </Text>
        </View>
        <View
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "paddingTop": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#3e4b57",
                  "fontFamily": "tit-light",
                  "fontSize": 17,
                  "opacity": 1,
                  "textAlign": "center",
                }
              }
            >
              Your current activity is not complete.
            </Text>
            <Text
              style={
                Object {
                  "color": "#3e4b57",
                  "fontFamily": "tit-light",
                  "fontSize": 17,
                  "opacity": 1,
                  "paddingBottom": 20,
                  "textAlign": "center",
                }
              }
            >
              Would you like to save it as is?
            </Text>
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "white",
                  "borderColor": "#CCC",
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "width": "80%",
                }
              }
            >
              <View
                style={
                  Object {
                    "color": "#3e4b57",
                    "fontFamily": "tit-light",
                    "height": 45,
                    "opacity": 0.36,
                    "width": "100%",
                  }
                }
              >
                <RNCPicker
                  items={
                    Array [
                      Object {
                        "label": "Select a reason...",
                        "textColor": 4291611852,
                        "value": "placeholder",
                      },
                      Object {
                        "label": "Out of breath",
                        "textColor": 4282272599,
                        "value": "out_of_breath",
                      },
                      Object {
                        "label": "Out of time",
                        "textColor": 4282272599,
                        "value": "out_of_time",
                      },
                      Object {
                        "label": "Feeling lazy",
                        "textColor": 4282272599,
                        "value": "feeling_lazy",
                      },
                      Object {
                        "label": "Other",
                        "textColor": 4282272599,
                        "value": "other",
                      },
                    ]
                  }
                  onChange={[Function]}
                  selectedIndex={0}
                  style={
                    Array [
                      Object {
                        "height": 216,
                      },
                      undefined,
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-around",
              "marginTop": 20,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "width": "40%",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "white",
                    "borderColor": "#CCC",
                    "borderRadius": 8,
                    "borderWidth": 1,
                    "justifyContent": "center",
                    "paddingHorizontal": 14,
                    "paddingVertical": 8,
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 2,
                      "width": 0,
                    },
                    "shadowOpacity": 0.25,
                    "shadowRadius": 2.84,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#3e4b57",
                      "fontFamily": "tit-regular",
                      "fontSize": 14,
                      "opacity": 1,
                      "textAlign": "center",
                      "textTransform": "uppercase",
                    }
                  }
                >
                  No
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "width": "40%",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "white",
                    "borderColor": "#CCC",
                    "borderRadius": 8,
                    "borderWidth": 1,
                    "justifyContent": "center",
                    "paddingHorizontal": 14,
                    "paddingVertical": 8,
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 2,
                      "width": 0,
                    },
                    "shadowOpacity": 0.25,
                    "shadowRadius": 2.84,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#3e4b57",
                      "fontFamily": "tit-regular",
                      "fontSize": 14,
                      "opacity": 1,
                      "textAlign": "center",
                      "textTransform": "uppercase",
                    }
                  }
                >
                  Yes
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
